src/node_buffer.cc

```diff
@@ -219,7 +219,10 @@ MaybeLocal<Object> New(Isolate* isolate,
   CHECK(actual <= length);
 
   if (actual < length) {
-    data = static_cast<char*>(realloc(data, actual));
+    char* new_data = static_cast<char*>(realloc(data, actual));
```

Comment:
because whenever realloc fails and it returns Null , whenever this happens  its old value ( data ) leaks.